### PR TITLE
fs/sync: add sync api

### PR DESCRIPTION
**Pull request by Donny9**
## Summary

sync: add sync and syncfs api

## Impact

support sync

## Testing

Ci

### CODE REVIEW

**Comment by yamt** [comment]
do you have a plan to actually implement them?

**Comment by Donny9** [comment]
> do you have a plan to actually implement them?

No, add them for iozone tools only.

**Comment by Donny9** [comment]
> 

There is currently no good idea to implement them, so assume they will always succeed

**Comment by pkarashchenko** [comment]
Then maybe it is more easy to handle this at iozone tools integration level only?

File: libs/libc/unistd/lib_sync.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *   syncfs() returns 0 on success; on error, it returns -1 and sets
+ *   errno to indicate the error.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int syncfs(int fd)
+{
+  return 0;
```
Comment (xiaoxiang781216) [inline libs/libc/unistd/lib_sync.c:84]: @Donny9 we can forward to fsync directly

File: libs/libc/unistd/lib_sync.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * Returned Value:
+ *   sync() is always successful.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+void sync(void)
+{
```
Comment (xiaoxiang781216) [inline libs/libc/unistd/lib_sync.c:64]: maybe we can iterate through all process and then file table

File: libs/libc/unistd/lib_sync.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/unistd/lib_sync.c:84]: Here is the patch: https://github.com/apache/nuttx/pull/8092

**Comment by xiaoxiang781216** [comment]
No, sync isn't implemented yet.

File: syscall/syscall.csv [hunk 3]
```diff
@@ -165,6 +165,7 @@
 "socketpair","sys/socket.h","defined(CONFIG_NET)","int","int","int","int","int [2]|FAR int *"
 "stat","sys/stat.h","","int","FAR const char *","FAR struct stat *"
 "statfs","sys/statfs.h","","int","FAR const char *","FAR struct statfs *"
+"sync","unistd.h","","void"
```
Comment (xiaoxiang781216) [inline syscall/syscall.csv:168]: move after line 169

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 @Donny9 can we close this PR?

@pkarashchenko sync is ready for review now.

**Review by xiaoxiang781216 — APPROVED** [review]

File: syscall/syscall.csv (hunk 3 above)
Comment (Donny9) [reply syscall/syscall.csv]: Done!

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**